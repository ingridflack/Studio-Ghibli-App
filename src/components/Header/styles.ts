import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
  `}
`;

export const Logo = styled.img`
  width: 8rem;
`;

export const LinksContainer = styled.div``;

export const NavLink = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.gray.white};
    transition: border-bottom, color 300ms ease;
    border-bottom: 1px solid black;
    padding: 0.25rem 0;

    & + & {
      margin-left: 1rem;
    }

    &:hover {
      color: ${theme.colors.gray.lightGray};
      border-color: ${theme.colors.gray.lightGray};
    }

    @media screen and (max-width: 576px) {
      justify-self: flex-end;
    }
  `}
`;
