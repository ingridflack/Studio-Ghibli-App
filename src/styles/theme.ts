const theme = {
  colors: {
    background: "#000",

    gray: {
      white: "#f6f6f6",
      platinum: "#e1e1e6",
      silver: "#a8a8b3",
      lightGray: "#7b8c98",
      dark: "#141414",
    },

    primary: "#e50914",
    secondary: "#b20710",
  },
  fontSize: {
    tiny: "0.65rem",
    small: "0.75rem",
    medium: "0.875rem",
    normal: "1rem",
    large: "1.5rem",
  },
} as const;

export type Theme = typeof theme;

export default theme;
