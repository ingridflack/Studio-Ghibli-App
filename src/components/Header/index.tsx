import Link from "next/link";

import * as S from "./styles";

const Header = () => {
  return (
    <S.Container>
      <S.Logo src="/assets/studio-ghibli-logo.webp" />

      <div>
        <Link href="/">
          <S.NavLink>Home</S.NavLink>
        </Link>
        <Link href="/favorites">
          <S.NavLink>Favorites</S.NavLink>
        </Link>
      </div>
    </S.Container>
  );
};

export default Header;
