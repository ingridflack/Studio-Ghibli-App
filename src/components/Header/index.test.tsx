import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Header from ".";
import theme from "../../styles/theme";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("Header", () => {
  it("should match snapshot", () => {
    useRouter.mockImplementationOnce(() => ({
      pathname: "/favorites",
    }));

    const { container } = render(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
