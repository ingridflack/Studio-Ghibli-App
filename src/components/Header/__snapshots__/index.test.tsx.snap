// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <div
    class="sc-acc38eb5-0 jBEmtF"
  >
    <img
      alt="Studio Ghibli logo"
      class="sc-acc38eb5-1 hxYLtv"
      height="72"
      src="/assets/studio-ghibli-logo.webp"
      width="128"
    />
    <nav>
      <a
        class="sc-acc38eb5-2 bYjima"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-acc38eb5-2 iwGSMA"
        href="/favorites"
      >
        Favorites
      </a>
    </nav>
  </div>
</div>
`;
